### PR TITLE
refactor: 代码更优雅

### PR DESCRIPTION
**Pull request by 2nthony**
#23 

## Description

重写脚本

- kleur 彩色字体
- glob 读取`templates`文件
- fs-extra 好用的`fs`工具库

### [SAO](https://saojs.org)

- 参考 logger
- 参考 生成,移动文件的方式

## Screenshot

Screen Shot 2019-04-26 at 11 15 44 PM

## Checklist:

- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly. (PR: #)
- [ ] I have added tests to cover my changes (if not applicable, please state why)
- [ ] All new and existing tests are passing.

### CODE REVIEW

File: bin/cli.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+  fs.outputFileSync(target, content)
+
+  logger.fileAction('magenta', 'Created', path.relative(process.cwd(), target))
+})
+
+/**
+ * Moved template files
+ */
+const patterns = {
```
Comment (comeontom) [inline bin/cli.js]: rename it, dont need comment

File: bin/cli.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  '✍️  What is the npm name of your component? '
+)
+
+const componentName = kebabcasify(npmName)
+const outDir = path.join(process.cwd(), componentName)
+
+/**
+ * Created template files
+ */
+templates.forEach(filepath => {
```
Comment (comeontom) [inline bin/cli.js]: let this become a fucntion, so you just need a meaningful function, not a comment

File: bin/cli.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+/**
+ * Moved template files
+ */
+const patterns = {
+  gitignore: '.gitignore',
+  'package-json': 'package.json',
+  'src/component.vue': `src/${componentName}.vue`
+}
+Object.keys(patterns).forEach(pattern => {
```
Comment (comeontom) [inline bin/cli.js]: like above, here become a function

**Comment by comeontom** [comment]
I dont see the relationship with sao🤔?

**Comment by 2nthony** [comment]
> I dont see the relationship with sao🤔?

https://github.com/FEMessage/vue-sfc-cli/commit/c5b9f4b93246e2b6dec304f32a4013554d0b0a4e

`logger.js` is modified from `sao/lib/logger.js`

File: bin/cli.js [hunk 4]
```diff
@@ -21,43 +16,19 @@ const npmName = readline.question(
...
-
-  fs.outputFileSync(target, content)
-
-  logger.fileAction('magenta', 'Created', path.relative(process.cwd(), target))
+const fileActions = new FileActions({
+  componentName,
+  outDir,
+  templatesDir: path.join(__dirname, '../templates'),
+  patterns: {
+    gitignore: '.gitignore',
```
Comment (levy9527) [inline bin/cli.js]: patterns是什么？

Comment (2nthony) [reply bin/cli.js]: 因为glob类工具都是用`pattern`来代表要匹配的字符

**Comment by 2nthony** [comment]
https://pnpm.js.org/

导图

image

**Pull request merged**